internal/cri/config/config.go

```diff
@@ -527,6 +530,10 @@ func ValidateRuntimeConfig(ctx context.Context, c *RuntimeConfig) ([]deprecation
 	}
 
 	for k, r := range c.ContainerdConfig.Runtimes {
+		if r.CgroupWritable && !opts.IsCgroup2UnifiedMode() {
+			return warnings, fmt.Errorf("runtime %s: `cgroup_writable` is only supported on cgroup v2", k)
```

Comment:
Curious (sorry I don't have full context on the implementation, correct me if wrong):
- It looks like you're doing a validation based on platform (OS, cgroup v1 vs v2, etc) and config;
- A common case might be, users would like to enable writable cgroup at best effort, which means they expect a **fallback** (no-op) instead of **failure**;

Reply:
Thanks for the reply! A second thought I think that makes sense.